### PR TITLE
[opt](nereids)adjust aggregation cost to make it compatible to the cost of join and shuffle 

### PR DESCRIPTION
**Pull request by englefly**
## Proposed changes
1. set column stats for count() agg function to Unknown
2. stats derive for having clause is adjusted. before, we use experenced factor 0.01. Now it is estimated as normal filter
3. change DEFAULT_AGGREGATE_RATIO from 0.5 to 1/3
4. support edge case for unionColumnStatistics when any side is Unknown, the result is unknown
5. skip Unkown  column stats in Statistics.enforceValid()
6. dataSizeFactor() if all col stats of visible slots are unkown, set the lower/upperbound (0.03, 0.07)

release cluster 10BE
tpcds 10t
before 3057sec, after 3040sec

before: 3057 sec
query1	5521	3314	4581	3314
query2	20157	19047	19545	19047
query3	15812	22905	22734	22734
query4	116065	109370	116286	109370
query5	39214	38590	47209	38590
query6	9620	8232	8759	8232
query7	33301	37778	35619	35619
query8	9441	7939	10316	7939
query9	164140	169847	173415	169847
query10	21638	27476	27143	27143
query11	60017	60918	62195	60918
query12	5826	4563	4664	4563
query13	40385	46801	49380	46801
query14	152567	150127	155888	150127
query15	6220	6251	4885	4885
query16	24634	36006	19540	19540
query17	43044	37558	31283	31283
query18	25276	21139	22313	21139
query19	25106	26419	29907	26419
query20	10267	6008	10332	6008
query21	9307	3411	2934	2934
query22	3711	3319	3318	3318
query23	262386	263307	264084	263307
query24	37475	38317	35238	35238
query25	32888	37608	34260	34260
query26	15592	18622	17922	17922
query27	31244	42411	41540	41540
query28	141031	141499	134866	134866
query29	49133	50908	49850	49850
query30	7119	4715	3346	3346
query31	12503	12732	11773	11773
query32	7847	9097	9520	9097
query33	32739	52729	32124	32124
query34	13677	19271	17114	17114
query35	25557	25084	28002	25084
query36	20629	20581	21652	20581
query37	11944	11680	15140	11680
query38	24456	24778	24408	24408
query39	6370	6226	5707	5707
query40	7915	15226	8983	8983
query41	895	486	463	463
query42	22312	18923	27068	18923
query43	8106	8406	7463	7463
query44	59617	59106	60050	59106
query45	7202	5928	6285	5928
query46	20796	19904	21178	19904
query47	20069	19538	19480	19480
query48	27015	25586	25766	25586
query49	47576	45531	50639	45531
query50	19361	18131	18091	18091
query51	22185	19395	19730	19395
query52	16939	20730	20165	20165
query53	25739	25084	25184	25084
query54	27170	28023	26363	26363
query55	14617	13586	13886	13586
query56	32906	34828	30251	30251
query57	11332	10860	10901	10860
query58	26999	29529	35555	29529
query59	24765	25181	24145	24145
query60	33905	33790	34612	33790
query61	61628	57307	55993	55993
query62	5666	5392	5980	5392
query63	28810	29030	30400	29030
query64	42222	40727	65072	40727
query65	33951	33274	32996	32996
query66	15410	16598	19878	16598
query67	116222	116734	112534	112534
query68	29394	34063	31045	31045
query69	26376	24911	28309	24911
query70	15846	18986	16639	16639
query71	26249	27143	29661	27143
query72	170	163	159	159
query73	13519	13816	13342	13342
query74	35905	34898	36680	34898
query75	41561	41124	41236	41124
query76	28142	31558	28384	28384
query77	33883	28975	31949	28975
query78	74649	113090	114924	113090
query79	23562	22963	22637	22637
query80	57645	55937	54879	54879
query81	9027	4546	4001	4001
query82	30456	28888	18990	18990
query83	8726	7859	5473	5473
query84	9886	2569	10050	2569
query85	17076	11273	9113	9113
query86	4842	4779	4518	4518
query87	26629	26398	26718	26398
query88	45880	45885	45108	45108
query89	24468	28838	29466	28838
query90	3791	14470	11761	11761
query91	12266	9182	6818	6818
query92	7058	6627	8129	6627
query93	28489	26211	27101	26211
query94	11315	18897	9364	9364
query95	30993	25833	23754	23754
query96	7983	6279	6184	6184
query97	36275	38188	37766	37766
query98	14681	23647	22776	22776
query99	10844	10725	10826	10725
Total cold run time: 3102775 ms
Total hot run time: 3057783 ms

after 3059 sec
query1	4559	3275	3566	3275
query2	18848	18784	19511	18784
query3	22311	22156	23540	22156
query4	114514	109474	110319	109474
query5	44811	58148	35134	35134
query6	11631	10018	9083	9083
query7	31580	31799	36181	31799
query8	9192	8669	8274	8274
query9	157570	154174	61765	61765
query10	17733	30374	25210	25210
query11	60245	61130	58896	58896
query12	7732	4607	5199	4607
query13	37433	54115	54186	54115
query14	135199	135486	134245	134245
query15	6715	4768	5422	4768
query16	16903	28680	32216	28680
query17	41850	32683	30896	30896
query18	23098	21658	22242	21658
query19	31639	26636	30911	26636
query20	5376	10867	6103	6103
query21	8495	2785	4359	2785
query22	3760	3737	3551	3551
query23	264060	258855	264446	258855
query24	39533	39318	35202	35202
query25	32060	31363	32018	31363
query26	15765	14336	16438	14336
query27	37050	44031	47109	44031
query28	144738	146493	146770	146493
query29	48873	57416	48768	48768
query30	7416	4006	3548	3548
query31	13062	11585	12092	11585
query32	6564	6343	5894	5894
query33	35354	37197	33609	33609
query34	13837	12701	12853	12701
query35	23706	23610	25498	23610
query36	20319	19597	20699	19597
query37	9153	10540	9430	9430
query38	26903	24213	25445	24213
query39	7403	5802	5690	5690
query40	7192	6713	9986	6713
query41	1141	802	448	448
query42	19819	21061	20148	20148
query43	8290	7265	8088	7265
query44	57570	54795	57409	54795
query45	5561	6083	4374	4374
query46	20027	19636	20500	19636
query47	19948	19521	19383	19383
query48	25115	26412	30895	26412
query49	48725	56716	45787	45787
query50	19554	17505	18479	17505
query51	19522	19973	20381	19973
query52	17017	20722	23037	20722
query53	28236	26989	27023	26989
query54	24224	26760	29639	26760
query55	17575	15594	14232	14232
query56	34584	73754	33905	33905
query57	11578	10983	11510	10983
query58	34545	34361	32886	32886
query59	24736	24199	24820	24199
query60	33267	35164	46649	35164
query61	69805	59957	54374	54374
query62	5705	5234	5893	5234
query63	25949	25264	25390	25264
query64	42141	42313	43791	42313
query65	33707	33175	33913	33175
query66	15765	16585	17916	16585
query67	117101	113159	113392	113159
query68	29726	28004	29223	28004
query69	29079	28005	26951	26951
query70	15543	15818	16449	15818
query71	27782	28750	29451	28750
query72	173	160	158	158
query73	11479	12259	12920	12259
query74	35979	35810	33981	33981
query75	54487	54417	54914	54417
query76	28332	30736	34253	30736
query77	34140	32821	34632	32821
query78	78448	115208	114110	114110
query79	21112	20870	22570	20870
query80	54250	52891	53305	52891
query81	10335	4219	3987	3987
query82	25262	18677	18569	18569
query83	7053	8318	16596	8318
query84	8190	3363	3043	3043
query85	17596	14204	18524	14204
query86	5260	4569	6338	4569
query87	26375	26709	26144	26144
query88	44883	44517	45421	44517
query89	21123	27995	24258	24258
query90	13886	10167	11079	10167
query91	8981	3777	3861	3777
query92	4338	4066	5427	4066
query93	28991	26479	26210	26210
query94	8936	19945	18665	18665
query95	27332	27977	20673	20673
query96	7330	7394	5927	5927
query97	36101	37612	37980	37612
query98	14861	22531	21610	21610
query99	11746	11037	11072	11037
Total cold run time: 3094498 ms
Total hot run time: 2948321 ms
query9 偶发报错 s3 读超时，所以q9以成功执行的时间计算，总时间是3040sec

Issue Number: close #xxx

### CODE REVIEW

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run cloud_p0

**Review by starocean999 — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java [hunk 1]
```diff
@@ -329,14 +329,16 @@ public Cost visitPhysicalHashAggregate(
             PhysicalHashAggregate<? extends Plan> aggregate, PlanContext context) {
         Statistics inputStatistics = context.getChildStatistics(0);
         double exprCost = expressionTreeCost(aggregate.getExpressions());
+        double aggCostBaseFactor = 3;
```
Comment (xzj7019) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java:332]: current cost model is sensitive, do we need to do full test on all sf benchmark and user cases?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java [hunk 2]
```diff
@@ -164,24 +160,6 @@ public Statistics visitComparisonPredicate(ComparisonPredicate cp, EstimationCon
         }
         ColumnStatistic statsForLeft = ExpressionEstimation.estimate(left, context.statistics);
         ColumnStatistic statsForRight = ExpressionEstimation.estimate(right, context.statistics);
-        if (aggSlots != null) {
```
Comment (xzj7019) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java:167]: means we give up the estimation of having?

Comment (xzj7019) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/stats/FilterEstimation.java:167]: pls add stats. estimation behavior changes notes in "Proposed changes", including fixed parameter changing, etc.

File: fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java (hunk 1 above)
Comment (xzj7019) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/cost/CostModelV1.java:332]: better to measure the influence of this change on ds 10t.

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Comment by englefly** [comment]
run buildall

**Review by starocean999 — DISMISSED** [review]

**Review by xzj7019 — APPROVED** [review]

**Comment by englefly** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Pull request merged**